[PG NCCL] Provide work obj in postProcess

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #100783
* #100782
* __->__ #100781
* #100780

This will be needed for allreduce_sparse so we can set the outputs on
the work object.

Differential Revision: [D45632428](https://our.internmc.facebook.com/intern/diff/D45632428/)

rohan-varma committed:
[PG NCCL] Provide work obj in postProcess

This will be needed for allreduce_sparse so we can set the outputs on
the work object.

Differential Revision: [D45632428](https://our.internmc.facebook.com/intern/diff/D45632428/)

[ghstack-poisoned]

fegin reviewed (approved):
Linter failure is real. Please fix linter before landing.

rohan-varma committed:
Update on "[PG NCCL] Provide work obj in postProcess"


This will be needed for allreduce_sparse so we can set the outputs on
the work object.

Differential Revision: [D45632428](https://our.internmc.facebook.com/intern/diff/D45632428/)

[ghstack-poisoned]
